api/db/services/document_service.py

```
@@ -506,6 +506,18 @@ def remove_document(cls, doc, tenant_id):
                 f"Failed to prune dataset_nav for document {doc.id}: {e}",
             )
 
+        # Record doc deletion for incremental structure-merge ghost cleanup
+        try:
+            from rag.svr.task_executor_refactor.dataset_structure_merger import (
+                record_doc_deletion,
+            )
+
+            record_doc_deletion(tenant_id, doc.kb_id, doc.id)
+        except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)